internal/lang/evalchecks/eval_for_each.go

```diff
@@ -135,6 +135,7 @@ func performTypeAndValueChecks(expr hcl.Expression, hclCtx *hcl.EvalContext, all
 
 }
 
+// TODO ephemeral - check how ephemeral should impact this function. Check also the unit tests
```

Comment:
Good question.  My initial thought is that it should follow the same conventions as Sensitive and only be allowed in each.value and not in each.key.